packages/std/src/ibc/transfer_msg_builder_v2.rs

```diff
@@ -0,0 +1,409 @@
+        TransferMsgBuilderV2 {
+            transfer_type: self.transfer_type,
+            to_address: self.to_address,
+            tokens: self.tokens,
+            memo: self.memo.add_src_callback(src_callback),
+        }
+    }
+}
+
+impl<Memo: AddDstCallbackPossible, TransferType> TransferMsgBuilderV2<Memo, TransferType> {
+    /// Adds an IBC destination callback entry to the memo field.
+    /// Use this if you want to receive IBC callbacks on the destination chain.
+    ///
+    /// For more info check out [`crate::IbcDestinationCallbackMsg`].
+    pub fn with_dst_callback(
+        self,
+        dst_callback: IbcDstCallback,
+    ) -> TransferMsgBuilderV2<Memo::CallbackType, TransferType> {
+        TransferMsgBuilderV2 {
+            transfer_type: self.transfer_type,
+            to_address: self.to_address,
+            tokens: self.tokens,
+            memo: self.memo.add_dst_callback(dst_callback),
+        }
+    }
+}
+
+impl<Memo: ForwardingPossible> TransferMsgBuilderV2<Memo, EmptyTransferType> {
+    /// Adds forwarding data.
+    /// It is worth to notice that the builder does not allow to add forwarding data along with
+    /// source callback. It is discouraged in the IBC docs:
+    /// https://ibc.cosmos.network/v9/middleware/callbacks/overview/#known-limitations
+    pub fn with_direct_transfer(
+        self,
+        channel_id: String,
+        ibc_timeout: IbcTimeout,
+    ) -> TransferMsgBuilderV2<Memo, Direct> {
+        TransferMsgBuilderV2 {
+            transfer_type: Direct {
+                transfer_type: TransferV2Type::Direct {
+                    channel_id,
+                    ibc_timeout,
+                },
+            },
+            to_address: self.to_address,
+            tokens: self.tokens,
+            memo: self.memo,
+        }
+    }
+
+    /// Adds forwarding data.
+    /// It is worth to notice that the builder does not allow to add forwarding data along with
+    /// source callback. It is discouraged in the IBC docs:
+    /// https://ibc.cosmos.network/v9/middleware/callbacks/overview/#known-limitations
+    pub fn with_forwarding(
+        self,
+        channel_id: String,
+        hops: Vec<Hop>,
+        timeout: Timestamp,
+    ) -> TransferMsgBuilderV2<Memo, Forwarding> {
+        TransferMsgBuilderV2 {
+            transfer_type: Forwarding {
+                transfer_type: TransferV2Type::MultiHop {
+                    channel_id,
+                    hops,
+                    timeout,
+                },
+            },
+            to_address: self.to_address,
+            tokens: self.tokens,
+            memo: self.memo,
+        }
+    }
+
+    /// Adds forwarding data.
+    /// It is worth to notice that the builder does not allow to add forwarding data along with
+    /// source callback. It is discouraged in the IBC docs:
+    /// https://ibc.cosmos.network/v9/middleware/callbacks/overview/#known-limitations
+    pub fn with_forwarding_unwinding(
```

Comment:
This has the same docs as `with_forwarding`. It would be nice to specify the difference between that function and this one in the doc comment.